### PR TITLE
different install dir for 32 bit remotevstplugin

### LINKED ISSUE #4717: `master`: Can't load 32bit VST plugins with 64bit Windows version of LMMS due to dll clashes
This is a bug introduced in #4421.
Previously `RemoteVstPlugin32.exe` was installed in a separate directory, with its dependencies(32bit DLLs). #4421 changed its location, so 32bit and 64bit `RemoteVstPlugin` are located in the same directory now.
According to the [MSDN documentation](https://docs.microsoft.com/en-us/windows/desktop/dlls/dynamic-link-library-search-order#search-order-for-desktop-applications), `RemoteVstPlugin32.exe` should be located in a directory different from where other remote plugin executables are. The directory should also contain required 32bit DLLs.

### PR DESCRIPTION
**Pull request by justnope**
remotevstplugin32 is installed in plugins/32. This should resolve #4717

### CODE REVIEW

File: plugins/vst_base/RemoteVstPlugin32.cmake [hunk 1]
```diff
@@ -1,32 +1,33 @@
 IF(LMMS_BUILD_WIN32 AND NOT LMMS_BUILD_WIN64)
 	ADD_SUBDIRECTORY(RemoteVstPlugin)
+	INSTALL(FILES "${MINGW_PREFIX}/bin/Qt5Core.dll" DESTINATION "${PLUGIN_DIR}/32")
+	INSTALL(FILES "${MINGW_PREFIX}/bin/zlib1.dll" DESTINATION "${PLUGIN_DIR}/32")
```
[the remark this answers is not in the document]
Comment (justnope) [reply plugins/vst_base/RemoteVstPlugin32.cmake]: indeed, thanks!

**Comment by PhysSong** [comment]
Tested MinGW 64bit build. Both 32bit and 64bit VSTs work fine with this patch.
I can't test it with MSVC now, but I guess @justnope can test the MSVC version.

**Comment by PhysSong** [comment]
I will merge this if there are no objections.

**Comment by karmux** [comment]
I can't load any VST plugins in Linux after this merge.
Error: Remote plugin '/home/karmo/lmms/build/plugins/32/RemoteVstPlugin32' not found.
How to solve this?

**Comment by PhysSong** [comment]
@karmux Thank yo for reporting! I'll fix the bug soon.

**Comment by karmux** [comment]
I did clean build and RemoteVstPlugin32 is still placed into plugins not plugins/32.

**Comment by PhysSong** [comment]
@karmux Are you running LMMS from the build directory? Now I think I only fixed the installation. I'll look into the problem.

**Comment by karmux** [comment]
@PhysSong I run from build directory.

**Comment by PhysSong** [comment]
@karmux Could you try this patch?
```diff
diff --git a/plugins/vst_base/CMakeLists.txt b/plugins/vst_base/CMakeLists.txt
index 314d5fc18..f0a896c4a 100644
--- a/plugins/vst_base/CMakeLists.txt
+++ b/plugins/vst_base/CMakeLists.txt
@@ -20,10 +20,9 @@ SET(EXTERNALPROJECT_ARGS
        BUILD_ALWAYS ON
 )
 
-SET(EXTERNALPROJECT_CMAKE_ARGS
+SET(EXTERNALPROJECT_CMAKE_ARGS_COMMON
        "-DLMMS_SOURCE_DIR=${CMAKE_SOURCE_DIR}"
        "-DLMMS_BINARY_DIR=${CMAKE_BINARY_DIR}"
-       "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY=${CMAKE_RUNTIME_OUTPUT_DIRECTORY}"
        "-DCMAKE_MODULE_PATH=${CMAKE_MODULE_PATH}"
        "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
 )
diff --git a/plugins/vst_base/RemoteVstPlugin32.cmake b/plugins/vst_base/RemoteVstPlugin32.cmake
index 07bed93b6..6f0ecfa23 100644
--- a/plugins/vst_base/RemoteVstPlugin32.cmake
+++ b/plugins/vst_base/RemoteVstPlugin32.cmake
@@ -1,3 +1,8 @@
+SET(EXTERNALPROJECT_CMAKE_ARGS_32
+       "${EXTERNALPROJECT_CMAKE_ARGS_COMMON}"
+       "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY=${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/32"
+)
+
 IF(LMMS_BUILD_WIN32 AND NOT LMMS_BUILD_WIN64)
        ADD_SUBDIRECTORY(RemoteVstPlugin)
        IF(MSVC)
@@ -8,7 +13,7 @@ IF(LMMS_BUILD_WIN32 AND NOT LMMS_BUILD_WIN64)
                INSTALL(FILES "${MINGW_PREFIX}/bin/Qt5Core.dll" DESTINATION "${PLUGIN_DIR}/32")
                INSTALL(FILES "${MINGW_PREFIX}/bin/zlib1.dll" DESTINATION "${PLUGIN_DIR}/32")
        ENDIF(MSVC)
-       INSTALL(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/../RemoteVstPlugin32.exe" DESTINATION "${PLUGIN_DIR}/32")
+       INSTALL(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/../32/RemoteVstPlugin32.exe" DESTINATION "${PLUGIN_DIR}/32")
 ELSEIF(LMMS_BUILD_WIN64 AND MSVC)
        SET(MSVC_VER ${CMAKE_CXX_COMPILER_VERSION})
 
@@ -41,12 +46,12 @@ ELSEIF(LMMS_BUILD_WIN64 AND MSVC)
                CMAKE_GENERATOR "${CMAKE_GENERATOR_32}"
                #CMAKE_GENERATOR_TOOLSET "${CMAKE_GENERATOR_TOOLSET}"
                CMAKE_ARGS
-                       "${EXTERNALPROJECT_CMAKE_ARGS}"
+                       "${EXTERNALPROJECT_CMAKE_ARGS_32}"
                        "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}"
                        "-DCMAKE_PREFIX_PATH=${QT_32_PREFIX}"
        )
 
-       INSTALL(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/../RemoteVstPlugin32.exe" DESTINATION "${PLUGIN_DIR}/32")
+       INSTALL(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/../32/RemoteVstPlugin32.exe" DESTINATION "${PLUGIN_DIR}/32")
 
        #TODO: find a solution when not using vcpkg for qt
        SET(VCPKG_ROOT_32 "${CMAKE_FIND_ROOT_PATH}/../x86-windows")
@@ -61,24 +66,24 @@ ELSEIF(LMMS_BUILD_LINUX)
        ExternalProject_Add(RemoteVstPlugin32
                "${EXTERNALPROJECT_ARGS}"
                CMAKE_ARGS
-                       "${EXTERNALPROJECT_CMAKE_ARGS}"
+                       "${EXTERNALPROJECT_CMAKE_ARGS_32}"
                        "-DCMAKE_CXX_COMPILER=${WINEGCC}"
                        "-DCMAKE_CXX_FLAGS=-m32"
        )
 
-       INSTALL(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/../RemoteVstPlugin32" "${CMAKE_CURRENT_BINARY_DIR}/../RemoteVstPlugin32.exe.so" DESTINATION "${PLUGIN_DIR}/32")
+       INSTALL(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/../32/RemoteVstPlugin32" "${CMAKE_CURRENT_BINARY_DIR}/../32/RemoteVstPlugin32.exe.so" DESTINATION "${PLUGIN_DIR}/32")
 
 ELSEIF(CMAKE_TOOLCHAIN_FILE_32)
        ExternalProject_Add(RemoteVstPlugin32
                "${EXTERNALPROJECT_ARGS}"
                CMAKE_ARGS
-                       "${EXTERNALPROJECT_CMAKE_ARGS}"
+                       "${EXTERNALPROJECT_CMAKE_ARGS_32}"
                        "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_32}"
                        "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE_32}"
        )
        INSTALL(FILES "${CMAKE_PREFIX_PATH_32}/bin/Qt5Core.dll" DESTINATION "${PLUGIN_DIR}/32")
        INSTALL(FILES "${CMAKE_PREFIX_PATH_32}/bin/zlib1.dll" DESTINATION "${PLUGIN_DIR}/32")
-       INSTALL(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/../RemoteVstPlugin32.exe" DESTINATION "${PLUGIN_DIR}/32")
+       INSTALL(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/../32/RemoteVstPlugin32.exe" DESTINATION "${PLUGIN_DIR}/32")
 ELSE()
        MESSAGE(WARNING "Can't build RemoteVstPlugin32, unknown environment. Please supply CMAKE_TOOLCHAIN_FILE_32 and optionally CMAKE_PREFIX_PATH_32")
        RETURN()
diff --git a/plugins/vst_base/RemoteVstPlugin64.cmake b/plugins/vst_base/RemoteVstPlugin64.cmake
index f802bc4b9..2fe6ea185 100644
--- a/plugins/vst_base/RemoteVstPlugin64.cmake
+++ b/plugins/vst_base/RemoteVstPlugin64.cmake
@@ -1,3 +1,8 @@
+SET(EXTERNALPROJECT_CMAKE_ARGS_64
+       "${EXTERNALPROJECT_CMAKE_ARGS_COMMON}"
+       "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY=${CMAKE_RUNTIME_OUTPUT_DIRECTORY}"
+)
+
 IF(LMMS_BUILD_WIN64)
        ADD_SUBDIRECTORY(RemoteVstPlugin)
        INSTALL(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/../RemoteVstPlugin64.exe" DESTINATION "${PLUGIN_DIR}")
@@ -5,7 +10,7 @@ ELSEIF(LMMS_BUILD_LINUX)
        ExternalProject_Add(RemoteVstPlugin64
                "${EXTERNALPROJECT_ARGS}"
                CMAKE_ARGS
-                       "${EXTERNALPROJECT_CMAKE_ARGS}"
+                       "${EXTERNALPROJECT_CMAKE_ARGS_64}"
                        "-DCMAKE_CXX_COMPILER=${WINEGCC}"
                        "-DCMAKE_CXX_FLAGS=-m64"
        )
```

**Comment by karmux** [comment]
@PhysSong this patch fixes it :+1:

**Pull request merged**